bot.go

```diff
@@ -68,6 +78,9 @@ func New(h *Handlers) *Bot {
 	go b.processMessages()
 
 	b.startPeriodicCommands()
+
+	// store the Bot in a singleton to hand it back to anyone else who comes asking
+	singleton = b
```

Comment:
This will not work as expected, for example, I run multiple protocols, the last bot created would overwrite the first: https://github.com/fabioxgn/go-bot-heroku/blob/master/main.go#L30

Why do you need the bot instance? I thing that this should be exposed by the protocols, not the bot itself, which is a library and should not be used directly.

Reply:
Thanks for the response.

The itch I'm trying to scratch is #97.  What's the proper mechanism to receive an event that's not from a channel/message (in this case a webhook) and then publish a message to a channel?